src/main/kotlin/org/rust/ide/annotator/AnnotationSessionEx.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.annotator
+
+import com.intellij.lang.annotation.AnnotationSession
+import com.intellij.openapi.util.Key
+import com.intellij.util.containers.orNull
+import org.rust.lang.core.crate.Crate
+import org.rust.lang.core.psi.RsFile
+import org.rust.openapiext.getOrPut
+import java.util.*
+
+private val CRATE_KEY = Key<Optional<Crate>>("org.rust.ide.annotator.currentCrate")
+
+fun AnnotationSession.currentCrate(): Crate? {
```

Comment:
It is a bit long to write `holder.currentAnnotationSession.currentCrate()`, maybe also add helper method?
```kotlin
fun AnnotationHolder.currentCrate(): Crate? = currentAnnotationSession.currentCrate()
```